Slim/Formats/Playlists/CUE.pm

```diff
@@ -337,9 +860,118 @@ sub parse {
 		return;
 	}
 
+	my $msg = {};
+	$msg->{'message'}	= 'end of parse';
+	$msg->{'tracks'}		= $tracks;
+	main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump($msg));return $tracks;
+
 	return $tracks;
 }
 
+sub _addCommand{
+	my $cuesheet	= shift;
+	my $tracks		= shift;
+	my $inAlbum		= shift;
+	my $currtrack	= shift;
+	my $command		= shift;
+	my $value		= shift;
+
+	if ($inAlbum and !defined $cuesheet->{$command}){
+		$cuesheet->{$command} = $value;
+	} elsif (defined $currtrack and !defined $tracks->{$currtrack}->{$command}){
+		$tracks->{$currtrack}->{$command} = $value;
+	}
+
+	return ($cuesheet,$tracks);
+}
+
+sub _isRemCommandAccepted {
+
+	my $remCommand	= shift;
+	if (defined $refusedRemCommands{$remCommand}){
+		return 0;
+	}
+
+	# we don't want to accept from a REM a command to be ignored.
+	if (_isCommandToIgnore($remCommand)){
+		return 0;
+	}
+
+	return 1;
+}
+
+sub _isCommandAccepted{
+	my $command	= shift;
+
+	if (!_isStandardCommand($command)){
+		return 0;
+	}
+
+	if (_isCommandToIgnore($command)){
+		return 0;
+	}
+
+	return 1;
+}
+
+sub _isStandardCommand{
+	my $command	= shift;
+	return (defined $standardCueCommands{$command});
+}
+
+sub _isCommandToIgnore{
+	my $command	= shift;
+	return (defined $refusedCueCommands{$command});
+}
+
+sub _getCommandFromLine{
+	my $line	= shift;
+
+	if ($line =~ /^\s*(\S+)\s+(.*)/i){
+
+		return ($1,$2);
+	}
+
+	return (undef,undef);
+}
+
+sub _getRemCommandFromLine{
+	my $line	= shift;
+
+	if ($line =~ /^\"(.*)\"/i){ #"
+
+		return (undef,$1);
+
+	}elsif ($line =~ /^\s*(\S+)\s+(.*)/i){
+
+		return ($1,$2);
+	}
+
+	return (undef,undef);
+
+}
+
+sub _removeQuotes{
+	my $line	= shift;
+
+	$line =~ s/^\"(.*?)\".*/$1/i;#"
+	return $line;
+}
+
+sub _validateBoolean{
```

Comment:
This can be written much simpler:

```perl
sub _validateBoolean {
   my $value = shift;
   return defined $value && $value =~ /1|YES|Y/i;
}
```